### PR TITLE
Apply custom Checkstyle rules in IntelliJ

### PR DESCRIPTION
**Pull request by pugnascotia**
It turns out you can configure custom Checkstyle rules using the plugin
for IntelliJ, you just have to tell the plugin where to find the code.
Therefore, update the contributing docs with the necessary process.

Also fork the Checkstyle rule `MissingJavadocTypeCheck` so that we can
disable it for test classes. While it's useful to describe test cases in
test methods' Javadoc, this is somewhat less true for test classes
themselves.

### CODE REVIEW

**Comment by mark-vieira** [comment]
@elasticmachine update branch

**Comment by pugnascotia** [comment]
@elasticmachine run elasticsearch-ci/2

File: buildSrc/src/main/java/org/elasticsearch/gradle/checkstyle/MissingJavadocTypeCheck.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
+////////////////////////////////////////////////////////////////////////////////
```
Comment (mark-vieira) [inline buildSrc/src/main/java/org/elasticsearch/gradle/checkstyle/MissingJavadocTypeCheck.java:1]: Can we use a normal block comment here? I believe our [notice file generation](https://github.com/elastic/elasticsearch/blob/a92a647b9f17d1bddf5c707490a19482c273eda3/buildSrc/src/main/groovy/org/elasticsearch/gradle/NoticeTask.groovy#L113) relies on that comment style.

**Review by mark-vieira — APPROVED** [review]
One comment, otherwise LGTM.

**Comment by pugnascotia** [comment]
Backported to `7.x` in c7b5c25bf94

**Pull request merged**